simulator/event_manager.go

```diff
@@ -327,31 +333,11 @@ func (c *EventHistoryCollector) eventMatches(event types.BaseEvent) bool {
 
 // filePage copies the manager's latest events into the collector's page with Filter applied.
 func (c *EventHistoryCollector) fillPage(size int) {
```

Comment:
This seems to be thread-safe since the calling `SetCollectorPageSize()` holds a lock on the manager.